app/services/creates_mentor_discussion_post.rb

```diff
@@ -3,21 +3,22 @@ def self.create!(*args)
     new(*args).create!
   end
 
-  attr_reader :iteration, :user, :content, :discussion_post
-  def initialize(iteration, user, content)
-    super
+  attr_reader :mentor
+  def initialize(iteration, mentor, content)
+    @mentor = mentor
+    super(iteration, mentor, content)
```

Comment:
I really like the distinction between `author` and `mentor` rather than using `user` everywhere.